src/eradiate/scenes/phase/_rayleigh.py

```diff
@@ -19,26 +27,142 @@ class RayleighPhaseFunction(PhaseFunction):
     atmosphere.
     """
 
-    depolarization: float | None = documented(
+    depolarization: np.ndarray | t.Callable[[KernelContext], np.ndarray] = documented(
         attrs.field(
-            default=None,
-            converter=attrs.converters.optional(float),
+            converter=lambda x: x if callable(x) else np.array(x, dtype=np.float64),
+            kw_only=True,
+            default=np.array([0.0]),
```

Comment:
Avoid mutable defaults: here, you need a factory.
```suggestion
            factory=lambda: np.array([0.0]),
```